discovery-provider/plugins/notifications/src/email/notifications/index.ts

```diff
@@ -42,11 +40,33 @@ const Results = Object.freeze({
 export const getUsersCanNotify = async (
   identityDb: Knex,
   frequency: EmailFrequency,
-  startOffset: moment.Moment
-): Promise<EmailUsers> => {
+  startOffset: moment.Moment,
+  onPage: (emailUsers: EmailUsers) => Promise<void>,
+) => {
   // const validLastEmailOffset = startOffset.subtract(2, 'hours')
-  const userRows: { blockchainUserId: number; email: string }[] =
-    await identityDb
+  let lastUser: number = Number.MAX_SAFE_INTEGER; // query uses DESC, start at max
```

Comment:
query actually uses `ASC` by default for `userId`, `DESC` is for `timestamp`, no?
```
ORDER BY "userId", "timestamp" DESC
```
also the subsequent join doesn't preserve this order anyway, can you add an order by clause at the end of getUsersCanNotifyQuery after the joins